multiversion/src/main/scala/multiversion/commands/QueryBasedCommand.scala

```diff
@@ -0,0 +1,51 @@
+package multiversion.commands
+
+import java.nio.file.Path
+
+import moped.cli.Application
+import moped.cli.Command
+import moped.json.Result
+import moped.json.ValueResult
+import multiversion.diagnostics.MultidepsEnrichments._
+import os.Shellable
+
+/**
+ * A moped command that wraps around Bazel queries.
+ */
+abstract class QueryBasedCommand extends Command {
+  def app: Application
+  def bazel: Path
+  def bazelArgs: List[String]
+  def runCustomQuery(): Result[List[String]]
+
+  override def run(): Int = app.complete(runResult())
+
+  def runResult(): Result[Unit] =
+    for {
+      r <- runCustomQuery()
+    } yield r.foreach(Console.out.println(_))
+
+  /**
+   * Run Bazel query using moped and capture the output as text.
+   *
+   * @param queryExpression
+   */
+  protected def runQueryStr(queryExpression: String): Result[List[String]] = {
+    val command = List(
+      bazel.toString,
+      "query",
+      queryExpression,
+      "--notool_deps",
+      "--noimplicit_deps"
+    )
+    val lines =
+      app
+        .process((command ::: bazelArgs).map(x => (x: Shellable)): _*)
+        .call()
+        .out
+        .text()
+        .linesIterator
+        .toList
+    ValueResult(lines)
+  }
```

Comment:
There exists `BazelUtil.bazel` to run Bazel commands and display a progress bar for long running commands:
https://github.com/twitter/bazel-multiversion/blob/c09296ee8d36922fe088b909937b65effebcfabb/multiversion/src/main/scala/multiversion/BazelUtil.scala#L30-L35

There's also a helper to run `bazel query` in `LintCommand`:
https://github.com/twitter/bazel-multiversion/blob/c09296ee8d36922fe088b909937b65effebcfabb/multiversion/src/main/scala/multiversion/commands/LintCommand.scala#L100-L111

What do you think of moving `runQuery` to `BazelUtil` and reusing it?

Reply:
`runQuery` captures the output as Protobuf, but that doesn't work for `--output graph`, so I'm not using it.

In addition, `BazelUtil.bazel` adds a progress bar, which I didn't want for query wrappers. We can still move `runQueryStr` to `BazelUtil`.